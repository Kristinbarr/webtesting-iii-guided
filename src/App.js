<<<<<<< HEAD
import React, { Component } from 'react'
import logo from './logo.svg'

import Speak from './Speak'
import './App.css'
// import { resolve } from 'url';

export const asyncFunc = () => {
  return new Promise((resolve) => {
    setTimeout(() => {
      resolve('Success!')
    }, 1000)
  })
}
=======
import React, { Component } from "react";
import logo from "./logo.svg";
>>>>>>> 5cb5642b9147d654f866ca0a7ba87b39f7cc5fc4

import Speak from "./components/Speak";
import "./App.css";

export const asyncFunc = () => {
  return new Promise(resolve => {
    setTimeout(() => {
      resolve("Success!");
    }, 1000);
  });
};
class App extends Component {
  state = {
<<<<<<< HEAD
    message: ''
  }

  speak = () => {
    this.setState({ message: 'Bark' })
  }
=======
    message: ""
  };

  speak = () => {
    this.setState({ message: "Bark" });
  };
>>>>>>> 5cb5642b9147d654f866ca0a7ba87b39f7cc5fc4

  render() {
    return (
      <div className='App'>
        <header className='App-header'>
          <img src={logo} className='App-logo' alt='logo' />
          <p>
            Edit <code>src/App.js</code> and manually reload.
          </p>
          <a
            className='App-link'
            href='https://reactjs.org'
            target='_blank'
            rel='noopener noreferrer'
          >
            Learn React
          </a>
          <Speak speak={this.speak} message={this.state.message} />
        </header>
      </div>
    )
  }
}

export default App
