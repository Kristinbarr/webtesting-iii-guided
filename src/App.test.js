<<<<<<< HEAD
import React from 'react'
import renderer from 'react-test-renderer' // 1: install this npm module as a dev dependency
import { render, fireEvent } from '@testing-library/react'

import App, { asyncFunc } from './App'
=======
import React from "react";
import renderer from "react-test-renderer"; // 1: install this npm module as a dev dependency
import { render, fireEvent } from "@testing-library/react";

import App, { asyncFunc } from "./App";
>>>>>>> 5cb5642b9147d654f866ca0a7ba87b39f7cc5fc4

describe("<App />", () => {
  // 2. write this test
<<<<<<< HEAD
  it('matches snapshot', () => {
    const tree = renderer.create(<App />) // generates a DOM tree

    // snapshots are a JSON representation of the DOM tree
    expect(tree.toJSON()).toMatchSnapshot()
  })
})

describe('asyncFunc', () => {
  it('eventually resolves to success', () => {
    const expected = 'Success!'
    return expect(asyncFunc()).resolves.toBe(expected)
  })
})

describe('speak', () => {
  it("should pass 'bark' into Speak", () => {
    // getByText finds, queryByText is boolean
    const { getByTestId, queryByText } = render(<App />)

    expect(queryByText(/bark/i)).toBeFalsy()

    fireEvent.click(getByTestId(/speak/i)) // button element

    expect(queryByText(/bark/i)).toBeTruthy()
  })
})
=======
  it("matches snapshot", () => {
    const tree = renderer.create(<App />); // generates a DOM tree

    // snapshots are a JSON representation of the DOM tree
    expect(tree.toJSON()).toMatchSnapshot();
  });
});

describe("asyncFunc", () => {
  it("eventually resolves to success", () => {
    const expected = "Success!";
    return expect(asyncFunc()).resolves.toBe(expected);
  });
});

describe("speak", () => {
  it('Should pass "bark" into Speak', () => {
    const { getByTestId, queryByText } = render(<App />);

    expect(queryByText(/bark/i)).toBeFalsy();

    fireEvent.click(getByTestId(/speak/i));

    expect(queryByText(/bark/i)).toBeTruthy();
  });
});

// yarn add -D (npm install -D) @testing-library/react
>>>>>>> 5cb5642b9147d654f866ca0a7ba87b39f7cc5fc4
