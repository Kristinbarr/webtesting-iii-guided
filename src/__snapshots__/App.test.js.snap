// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <p>
      Edit 
      <code>
        src/App.js
      </code>
<<<<<<< HEAD
       and save to reload.
=======
       and manually reload.
>>>>>>> 5cb5642b9147d654f866ca0a7ba87b39f7cc5fc4
    </p>
    <a
      className="App-link"
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn React
    </a>
    <button
      data-testid="speak"
      onClick={[Function]}
    >
      Speak
    </button>
    <div>
      
    </div>
  </header>
</div>
`;
